solana/programs/swap-layer/src/processor/stage_outbound.rs

```diff
@@ -71,6 +71,12 @@ pub struct StageOutbound<'info> {
             // Cannot send to zero address.
             require!(args.recipient != [0; 32], SwapLayerError::InvalidRecipient);
 
+            // Min amount out must be specified for swaps into USDC.
+            require!(
+                args.min_amount_out.is_some() || src_mint.key() == common::USDC_MINT,
```

Comment:
So|| src_mint.key() == common::USDC_MINT will short-circuit this check. Is this relying on other constraints where the src mint and dst mint can't both be equal to 0?

Reply:
We want to short circuit the check when the mint is USDC. The user doesn't need to specify a min_amount_out in that case, since there won't be a swap.

Reply:
This check guarantees that min_amount_out is some when the src_mint is not USDC. We don't really care if someone specifies a min_amount_out when transferring USDC, since that value will be ignored.

Reply:
Hey sorry the original wording I had was very sloppy. Just edited it. Thanks for the clarification.